### DISCUSSION TITLE
How does the Milvus delete old data in MinIO ?

### DISCUSSION THREAD

**Discussion by kietheros** [milvus-io/milvus#31541] [Q&A and General discussion]
I install milvus standalone and create a collection (id: int64, feature_vector: float, dim 1280, index IVF_SQ8). I partition data by date and remove partitions which are older than 7 days. The collection now contains 3.5M rows. The milvus sizing tool estimates minio disk usage is 40GB x 4.
Here is the current disk usage:
+ minio/a-bucket 738 GB
+ rdb_data: 99GB

I'm concerned because Milvus is using too many disk storage compared to the estimation. Do you find these figures abnormal? How does the Milvus delete data in MinIO?

**Comment by yhmo** [comment]
The GC process is configured here:
https://github.com/milvus-io/milvus/blob/ab059bb064ea4be197044f267c168a7ec2d30516/configs/milvus.yaml#L433

gc.interval is the interval seconds that milvus triggers a GC task. Default value is 3600 seconds.
gc.missingTolerance is a configuration for orphan files, which means some orphan files might delay to be cleaned for 1 hour.
In runtime, a dropped collection/partition's data files might be kept in storage for 2~3 hours.
